twpca/model.py

```diff
@@ -260,12 +259,10 @@ def assign_warps(self, warps):
         # different warp intializations
         if warps is not None:
             # warps proved by user
-            if normalize_warps:
-                warps *= self.n_timepoints / np.max(warps)
             shift = warps[:, 0] - 1
             scale = np.ones(self.n_trials)
-            tau = np.hstack((np.ones((self.n_trials, 1)), np.diff(warps, axis=1)))
-            tau = np.maximum(0, tau) # make sure warps are monotonic
+            dt = np.maximum(0, np.diff(warps, axis=1)) # enforce monotonic increasing warps
+            tau = np.hstack((np.ones((self.n_trials, 1)), dt))
```

Comment:
`np.diff(...)` gives you one fewer tau than necessary (`shared_length - 1`), so you need to prepend something. If you prepend zeros you are in a much less friendly region of the softplus function (since you pass things backwards through an inverse softplus transform).